pkg/blockgen/blockgen_writer.go

```diff
@@ -37,9 +36,7 @@ import (
 // Note that the writer will not check if the target directory exists or
 // contains anything at all. It is the caller's responsibility to
 // ensure that the resulting blocks do not overlap etc.
-func NewBlockWriter(dir string) (Writer, error) {
-	logger := log.NewLogfmtLogger(os.Stderr)
-
+func NewBlockWriter(logger log.Logger, dir string) (Writer, error) {
 	res := &blockWriter{
 		logger: logger,
 		dir:    dir,
```

Comment:
Funny how I actually did remove logger arg on purpose here and you added back. Rational for original removal is this is the only place in the whole package that was taking logger. Nothing else. Felt kinda wrong. Also made the use of the package more brittle.

Are we not happy to create new default logger with `logger := log.NewLogfmtLogger(os.Stderr)`?

Reply:
I think creating new is worse than require-ing logging. This is because we end up with library (package) that has arbitrary logging. What if the caller uses totally different logger, want silence on `stderr` and want use JSON not fmt? We essentially block any of those by introducing new arbitrary logger.

In fact `log.Logger` is very simple interface that has just one method, so if anyone in the world want to use it, he/she can just implement their own, using go-kit logger or just pass `log.NewNoopLogger`.  
That's my thinking from the perspective of the caller of this.

Reply:
I merged, but happy to discuss it and if in later PRs.